### PR TITLE
Bug fix: Link operator names to the MLGraphBuilder methods

### PR DESCRIPTION
**Pull request by inexorabletash**
For a "reshape" reference a malformed IDL link was present as `*reshape*}}`. The other references are linked as appropriately to the builder references rather than just being styled text.

Documented this convention in SpecCodingConventions.md.

***
<a href="https://pr-preview.s3.amazonaws.com/inexorabletash/webnn/pull/547.html" title="Last updated on Feb 15, 2024, 12:36 AM UTC (e57c466)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/547/eb06ccf...inexorabletash:e57c466.html" title="Last updated on Feb 15, 2024, 12:36 AM UTC (e57c466)">Diff</a>

### CODE REVIEW

**Comment by zolkis** [comment]
Personally I'd prefer linking and it should be default generic thing, not a specific convention for this spec. 
This emphasis style looks like a leftover from early spec times with draft texts changing fast.

**Review by anssiko — APPROVED** [review]
Thanks, I support an update to the conventions & style guide doc.

**Comment by inexorabletash** [comment]
Updated with a note in the conventions doc.

File: index.bs [hunk 1]
```diff
@@ -1635,7 +1635,7 @@ partial interface MLGraphBuilder {
 <div class="note">
   <details open>
     <summary>
-    The behavior of this operation when the input tensor is 4-D of the {{MLInputOperandLayout/"nchw"}} layout and the activation is of operator type *relu* can be generically emulated from the usage of other operations as follow. However, user agents typically have a more efficient implementation for it, therefore its usage is encouraged from the performance standpoint.
+    The behavior of this operation when the input tensor is 4-D of the {{MLInputOperandLayout/"nchw"}} layout and the activation is of operator type {{MLGraphBuilder/relu()}} can be generically emulated from the usage of other operations as follow. However, user agents typically have a more efficient implementation for it, therefore its usage is encouraged from the performance standpoint.
```
Comment (huningxin) [inline index.bs:1638]: Does it make sense just saying "the activation is {{MLGraphBuilder/relu()}}"? I understand {{MLGraphBuilder/relu()}} is an `MLActivation` while previous text uses "operator type relu" to refer that.

File: index.bs [hunk 2]
```diff
@@ -2525,7 +2525,7 @@ partial interface MLGraphBuilder {
 </div>
 
 <div class="note">
-Although operations *greaterOrEqual* and *lesserOrEqual* can each be implemented in terms of operations *not*, *lesser*, and *greater* in other words `greater-or-equal(a, b)` is `not(lesser(a, b))`, they are specifically defined to handle NaN cases and for performance reason to avoid double comparisons.
+Although operations {{MLGraphBuilder/greaterOrEqual()}} and {{MLGraphBuilder/lesserOrEqual()}} can each be implemented in terms of operations {{MLGraphBuilder/not()}}, {{MLGraphBuilder/lesser()}}, and {{MLGraphBuilder/greater()}} in other words `greaterOrEqual(a, b)` is `not(lesser(a, b))`, they are specifically defined to handle NaN cases and for performance reason to avoid double comparisons.
```
Comment (huningxin) [inline index.bs:2528]: The spec usually uses `builder` in the emulation sample. Should we also be consistent in this compact emulation code? like `greaterOrEqual(a, b)` can be implemented by `builder.not(builder.lesser(a, b))`?

**Review by huningxin — APPROVED** [review]
LGTM with questions, thanks much!

File: index.bs (hunk 1 above)
Comment (inexorabletash) [reply index.bs]: Yes, done!

File: index.bs (hunk 2 above)
Comment (inexorabletash) [reply index.bs]: Makes sense - done!

File: index.bs [hunk 3]
```diff
@@ -3388,7 +3388,7 @@ partial interface MLGraphBuilder {
 <div class="note">
 <details open>
   <summary>
-    The behavior of this operation can be generically emulated via other operations as shown below, when the weight layout is the default {{MLGruWeightLayout/"zrn"}} layout, and the activation functions of the update/reset gate and new gate are of the operator types *sigmoid* and *tanh* respectively.
+    The behavior of this operation can be generically emulated via other operations as shown below, when the weight layout is the default {{MLGruWeightLayout/"zrn"}} layout, and the activation functions of the update/reset gate and new gate are of the operator types {{MLGraphBuilder/sigmoid()}} and {{MLGraphBuilder/tanh()}} respectively.
```
Comment (huningxin) [inline index.bs:3391]: Apologize I didn't mention in the last comment. We may also want to remove "of the operator types" here.

File: index.bs [hunk 4]
```diff
@@ -4397,7 +4397,7 @@ partial interface MLGraphBuilder {
 <div class="note">
 <details open>
   <summary>
-    The behavior of this operation can be generically emulated via other operations as shown below, when the weight layout is the default {{MLLstmWeightLayout/"iofg"}} layout, and the activation functions of the input/forget/output gate and the cell gate/the cell state's filter for the output hidden state are of the operator types *sigmoid* and *tanh* respectively.
+    The behavior of this operation can be generically emulated via other operations as shown below, when the weight layout is the default {{MLLstmWeightLayout/"iofg"}} layout, and the activation functions of the input/forget/output gate and the cell gate/the cell state's filter for the output hidden state are of the operator types {{MLGraphBuilder/sigmoid()}} and {{MLGraphBuilder/tanh()}} respectively.
```
Comment (huningxin) [inline index.bs:4400]: Same here, may remove "of the operator types".

File: index.bs (hunk 3 above)
Comment (inexorabletash) [reply index.bs:3391]: done!

File: index.bs (hunk 4 above)
Comment (inexorabletash) [reply index.bs:4400]: done! (and thanks for being explicit!)

**Comment by inexorabletash** [comment]
I noticed a few more potential places this style could be applied. Should I roll these in too, or leave them alone?

```diff
@@ -923,12 +923,12 @@ interface MLActivation {
 </div>
 
 <div class="note">
-These activations function types are used to create other operations. One such use of this interface is for when an activation function is fused into another operation such as [[#api-mlgraphbuilder-conv2d]] or [[#api-mlgraphbuilder-batchnorm]] during a graph construction session. Such fused activation functions can provide a significant performance improvement when supported natively by the underlying implementation. This is intended as an optimization opportunity for implementers.
+These activations function types are used to create other operations. One such use of this interface is for when an activation function is fused into another operation such as {{MLGraphBuilder/conv2d()}} or {{MLGraphBuilder/batchNormalization()}} during a graph construction session. Such fused activation functions can provide a significant performance improvement when supported natively by the underlying implementation. This is intended as an optimization opportunity for implementers.
 </div>
 
 ### Creating {{MLActivation}} ### {#api-mlactivation-create}
 <div class="note">
-The {{MLActivation}} objects (including the ones passed as input to methods) are created by the methods of {{MLGraphBuilder}} and are identified by their name. The |options| dictionary is defined by those methods. The actual creation of the activation function e.g. a [[#api-mlgraphbuilder-sigmoid-method]] or [[#api-mlgraphbuilder-relu-method]] can then be deferred until when the rest of the graph is ready to connect with it such as during the construction of [[#api-mlgraphbuilder-conv2d]] for example.
+The {{MLActivation}} objects (including the ones passed as input to methods) are created by the methods of {{MLGraphBuilder}} and are identified by their name. The |options| dictionary is defined by those methods. The actual creation of the activation function e.g. a {{MLGraphBuilder/sigmoid()}} or {{MLGraphBuilder/relu()}} can then be deferred until when the rest of the graph is ready to connect with it such as during the construction of {{MLGraphBuilder/conv2d()}} for example.
 </div>
 
 <details open algorithm>
@@ -3674,7 +3674,7 @@ Create a named {{MLOperand}} based on a descriptor, that can be used as an input
 </details>
 
 ### instanceNormalization ### {#api-mlgraphbuilder-instancenorm}
-Normalize the input using [[Instance-Normalization]]. Unlike [[#api-mlgraphbuilder-batchnorm]] where the mean and variance values used in the normalization are computed across all the samples in the batch dimension while the model is trained, the mean and variance values used in the instance normalization are computed on the fly for each input feature of each individual sample in the batch.
+Normalize the input using [[Instance-Normalization]]. Unlike {{MLGraphBuilder/batchNormalization()}} where the mean and variance values used in the normalization are computed across all the samples in the batch dimension while the model is trained, the mean and variance values used in the instance normalization are computed on the fly for each input feature of each individual sample in the batch.
 
 <script type=idl>
 dictionary MLInstanceNormalizationOptions {
@@ -3776,7 +3776,7 @@ partial interface MLGraphBuilder {
 </div>
 
 ### layerNormalization ### {#api-mlgraphbuilder-layernorm}
-Normalize the input using [[Layer-Normalization]]. Unlike [[#api-mlgraphbuilder-batchnorm]] where the mean and variance values are computed across all the samples in the batch dimension while the model is trained, and in [[#api-mlgraphbuilder-instancenorm]] where the mean and variance values are computed on the fly for each input feature of each individual sample in the batch, the means and variance values of the layer normalization are computed on the fly across all the input features of each individual sample in the batch.
+Normalize the input using [[Layer-Normalization]]. Unlike {{MLGraphBuilder/batchNormalization()}} where the mean and variance values are computed across all the samples in the batch dimension while the model is trained, and in {{MLGraphBuilder/instanceNormalization()}} where the mean and variance values are computed on the fly for each input feature of each individual sample in the batch, the means and variance values of the layer normalization are computed on the fly across all the input features of each individual sample in the batch.
 
 <script type=idl>
 dictionary MLLayerNormalizationOptions {
```

**Comment by zolkis** [comment]
Roll'em.

**Pull request merged**